src/Jobs/Notifications/ProductNotificationJob.php

```diff
@@ -43,6 +46,19 @@ public function __construct(
 		parent::__construct( $action_scheduler, $monitor, $notifications_service );
 	}
 
+	/**
+	 * Override Product Notification adding Offer ID for deletions.
+	 *
+	 * @param array $args Arguments with the item id and the topic.
+	 */
+	protected function process_items( $args ): void {
+		if ( isset( $args['topic'] ) && isset( $args['item_id'] ) && $this->is_delete_topic( $args['topic'] ) ) {
+			$args['data'] = [ 'offer_id' => WCProductAdapter::get_google_product_offer_id( $this->get_slug(), $args['item_id'] ) ];
```

Comment:
The reason I used `WCProductAdapter::get_google_product_offer_id` is because when a product is deleted is there a case that the delete Notification runs after the product is completely deleted form the database. So we will lost access to those metas and the product itself as well. 

I would go for sending the OfferID and rely on Google to check if the product actually exists.

Reply:
Thanks for the clarification. We could add a comment there to explain that the `offer_id` might match the real offer ID or not, depending on whether the product has been synced by us or not. Also, I think it would be helpful to discuss this with Google so they understand that the offer ID isn't always reliable. Or, we could consider renaming it to `gla_offer_id`.